src/lib/dns.coffee

```diff
@@ -165,14 +142,12 @@ module.exports = (dnschain) ->
                                 return @sendErr res, NAME_RCODE.SERVFAIL, cb
 
             else if S(q.name).endsWith '.dns'
-                # TODO: right now we're doing a catch-all and pretending they asked
-                #       for namecoin.dns...
                 res.answer.push gIP2type(q.name,ttl,QTYPE_NAME[q.type])(gConf.get 'dns:externalIP')
                 @log.debug gLineInfo('cb|.dns'), {q:q, answer:res.answer}
                 @sendRes res, cb
             else
                 @log.debug gLineInfo("deferring request"), {q:q}
-                @oldDNSLookup req, (packet, code) =>
+                @dnschain.cache.resolveOldDNS req, (code, packet) =>
                     _.assign res, _.pick packet, ['edns_version', 'edns_options',
                         'edns', 'answer', 'authority', 'additional']
```

Comment:
> Are these the only fields that will ever be needed for icann lookups? or is it specific to DNS?

These are the only fields that should matter in a DNS response. The other ones that matter should already be in `res` (hopefully, you'll need to test it of course, and feel free to add new tests, for example in `test/redis.coffee`).